### ISSUE TITLE
Fedora solaar warnings

### ISSUE THREAD

**Issue by kloczek** [pwr-Solaar/Solaar#409]
solaar started from command line prints a lot ow warnings:

/usr/lib/python2.7/site-packages/solaar/gtk.py:34: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  __import__(module)
/usr/lib/python2.7/site-packages/solaar/ui/__init__.py:38: PyGIDeprecationWarning: Since version 3.11, calling threads_init is no longer needed. See: https://wiki.gnome.org/PyGObject/Threading
  GLib.threads_init()
/usr/lib/python2.7/site-packages/solaar/ui/notify.py:33: PyGIWarning: Notify was imported without specifying a version first. Use gi.require_version('Notify', '0.7') before import to ensure that the right version gets loaded.
  from gi.repository import Notify
/usr/lib/python2.7/site-packages/solaar/ui/action.py:36: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "name, label, tooltip, stock_id" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  action = Gtk.Action(name, label, label, None)
/usr/lib/python2.7/site-packages/solaar/ui/tray.py:148: PyGIWarning: AppIndicator3 was imported without specifying a version first. Use gi.require_version('AppIndicator3', '0.1') before import to ensure that the right version gets loaded.
  from gi.repository import AppIndicator3
/usr/lib/python2.7/site-packages/solaar/ui/window.py:220: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  p._title = Gtk.Label(' ')
/usr/lib/python2.7/site-packages/solaar/ui/window.py:110: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  p._scanning = Gtk.Label(_("Scanning") + '...')
/usr/lib/python2.7/site-packages/solaar/ui/window.py:114: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  bp.pack_start(Gtk.Label(' '), True, True, 0)
/usr/lib/python2.7/site-packages/solaar/ui/window.py:129: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  b._label = Gtk.Label(label_text)
/usr/lib/python2.7/site-packages/solaar/ui/window.py:175: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "homogeneous" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  bb = Gtk.ButtonBox(Gtk.Orientation.HORIZONTAL)
/usr/lib/python2.7/site-packages/solaar/ui/window.py:83: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  b = Gtk.Button(label) if label else Gtk.Button()
/usr/lib/python2.7/site-packages/solaar/ui/window.py:319: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "homogeneous" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  bottom_buttons_box = Gtk.ButtonBox(Gtk.Orientation.HORIZONTAL)

**Comment by fbruetting** [comment]
For me it’s:
```
$ solaar
Traceback (most recent call last):
  File "/usr/bin/solaar", line 44, in <module>
    solaar.gtk.main()
  File "/usr/lib/python2.7/site-packages/solaar/gtk.py", line 89, in main
    _require('gi.repository.Gtk', 'gir1.2-gtk-3.0')
  File "/usr/lib/python2.7/site-packages/solaar/gtk.py", line 34, in _require
    __import__(module)
  File "/usr/lib64/python2.7/site-packages/gi/importer.py", line 145, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/lib64/python2.7/site-packages/gi/importer.py", line 145, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/lib64/python2.7/site-packages/gi/importer.py", line 146, in load_module
    dynamic_module = load_overrides(introspection_module)
  File "/usr/lib64/python2.7/site-packages/gi/overrides/__init__.py", line 125, in load_overrides
    override_mod = importlib.import_module(override_package_name)
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/lib64/python2.7/site-packages/gi/overrides/Gdk.py", line 180, in <module>
    Gdk.EventType._2BUTTON_PRESS = getattr(Gdk.EventType, "2BUTTON_PRESS")
AttributeError: type object 'EventType' has no attribute '2BUTTON_PRESS'
```

**Comment by jasontibbitts** [comment]
I can't tell if this is solaar installed from Fedora packaging, or solaar installed from this git repository.

The solaar that's packaged is bound to be somewhat old.  I think everyone's waiting for an actual release in order to do a proper update.

**Comment by kloczek** [comment]
Just checked and looks like Fedora is used solaar from a quite old git hash

%global git_date 20150528
%global git_hash cf27328

My advice try to make new release ASAP (0.10.0 ?) even if few things are not finished.
Another advice: try to make more frequent minor releases :)
One minor release every month will probably increase as well feedback and stream of fixes with minimal your effort to make new changes :)

If you will try to make shortly new official release I promise t push this new version to Fedora rawhide git PR to push updated package ASAP :P
As rawhide it is devel Fedora version risk that it will be worse than current one s not substantial :P

Just in case here is URL to solaar fedora package git repo https://src.fedoraproject.org/rpms/solaar

**Comment by kloczek** [comment]
A .. yet another advice. When you will be choosing new version branch name please use solaar-<version>.
With such branch name, pulled tarball from git will have <package_name>-<version> base root source code directory inside. With such directory name, it is a bit easier make deb or rpm package :)

**Comment by Lekensteyn** [comment]
Good suggests regarding naming, will try to keep that in mind (on the other hand, given the limited development resources, I think that releases will just be done directly from master).

**Comment by kloczek** [comment]
Please let me know when you will decide to make new release :)
It is really better to have more releases. Each release is kind of stimulant to have more people focus on the project of more people. It is better as well from point of view of tracking down some new effects.

**Comment by kloczek** [comment]
Sorry asking .. any updates about new solaar release?

**Comment by jasontibbitts** [comment]
For the record, I had to mess with the Fedora package (because it wasn't building in rawhide) and so put a bit of work into it.  So rawhide is now updated to git HEAD, as well as migrated to python3.  The rawhide package won't install on the release branches but it will build.

I will happily consider making a test update for F28 but I'm generally very reluctant to push changes to any release branch.  I'm running this myself without issues but that's certainly no guarantee.  Anyone willing to test and give karma?
